src/main/java/spireMapOverhaul/zones/keymaster/KeymasterZone.java

```diff
@@ -0,0 +1,90 @@
+package spireMapOverhaul.zones.keymaster;
+
+import com.badlogic.gdx.graphics.Color;
+import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.helpers.PotionHelper;
+import com.megacrit.cardcrawl.potions.*;
+import com.megacrit.cardcrawl.powers.*;
+import com.megacrit.cardcrawl.rewards.RewardItem;
+import spireMapOverhaul.abstracts.AbstractZone;
+import spireMapOverhaul.zoneInterfaces.CombatModifyingZone;
+import spireMapOverhaul.zoneInterfaces.EncounterModifyingZone;
+import spireMapOverhaul.zoneInterfaces.RewardModifyingZone;
+import spireMapOverhaul.zoneInterfaces.ShopModifyingZone;
+
+import java.util.ArrayList;
+
+import static spireMapOverhaul.util.Wiz.atb;
+import static spireMapOverhaul.util.Wiz.forAllMonstersLiving;
+
+public class KeymasterZone extends AbstractZone implements CombatModifyingZone, RewardModifyingZone, ShopModifyingZone {
+    public static final String ID = "Keymaster";
+
+    public KeymasterZone() {
+        super(ID, Icons.MONSTER, Icons.SHOP);
+        this.width = 3;
+        this.height = 3;
+    }
+
+    @Override
+    public AbstractZone copy() {
+        return new KeymasterZone();
+    }
+
+    @Override
+    public Color getColor() {
+        return new Color(0.396f, 1.0f, 0.0f, 0.8f);
+    }
+
+    @Override
+    public boolean canSpawn() {
+        return (Settings.hasSapphireKey && Settings.hasEmeraldKey && Settings.hasRubyKey);
+    }
+
+    @Override
+    protected boolean canIncludeEarlyRows() {
+        return true;
+    }
+
+    @Override
+    public void atBattleStart() {
+        forAllMonstersLiving(m -> {
```

Comment:
How about we remove the enemy debuff and just give the player 1 focus as well.

That should be enough and fits themtically with the blue key

Reply:
I thought about that, but that point of Focus is useless on every base character besides Defect.

Maybe I should make it so that if the player is Defect, Focus is granted, otherwise the enemies just lose Strength?